test/integration/deployment/util.go

```diff
@@ -87,6 +94,40 @@ func newDeployment(name, ns string, replicas int32) *v1beta1.Deployment {
 	}
 }
 
+// newDeployment returns a RollingUpdate Deployment with with a fake container image
+func newAppsV1beta2Deployment(name, ns string, replicas int32) *v1beta2.Deployment {
```

Comment:
This and `deploymentAppsV1beta2Test` are not strictly necessary. You can use the extensions/v1beta1 client to create the Deployment, then test that you can update the selector through the extensions/v1beta2 client. You can then use the apps/v1beta2 client to Get() the same Deployment (remember it will auto-convert), and then use the apps/v1beta2 client to Update.